feature-buy-api/src/main/java/io/novafoundation/nova/feature_buy_api/domain/TradeTokenRegistry.kt

```diff
@@ -0,0 +1,56 @@
+package io.novafoundation.nova.feature_buy_api.domain
+
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+
+typealias TradeProvider = TradeTokenRegistry.Provider<*>
+
+interface TradeTokenRegistry {
+
+    fun hasProvider(chainAsset: Chain.Asset): Boolean
+
+    fun hasProvider(chainAsset: Chain.Asset, tradeFlow: TradeFlow): Boolean
+
+    fun availableProvidersFor(chainAsset: Chain.Asset, tradeFlow: TradeFlow): List<Provider<*>>
+
+    interface Provider<I : Integrator<*>> {
+        val id: String
+
+        val name: String
+
+        val officialUrl: String
+
+        val supportedFlows: Set<TradeFlow>
+
+        @get:DrawableRes
+        val logoRes: Int
+
+        @get:StringRes
+        val descriptionRes: Int
+
+        fun getPaymentMethods(tradeFlow: TradeFlow): List<PaymentMethod>
+
+        fun createIntegrator(chainAsset: Chain.Asset, address: String, tradeFlow: TradeFlow): I
+    }
+
+    interface Integrator<T> {
+
+        fun openFlow(using: T)
+    }
+
+    enum class TradeFlow {
```

Comment:
I suggest to rename it to `TradeType` or something, since when I first saw `TradeFlow` name I first though it it a typealias to something like `Flow<Trade>`